File: Event Management System/event-service/src/main/java/com/event/eventservice/service/EventService.java
package com.event.eventservice.service;

import com.event.eventservice.models.Event;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public interface EventService {
	Event createEvent(Event event);
	List<Event> getAllEvents();
	Event getEventById(Long id);
	Event updateEvent(Long id,Event event);
	void deleteEvent(Long id);
	void reduceSeats(Long id, int count);
	void increaseSeats(Long id, int count);
	void triggerFeedback(Long id);
	List<Event> getEventsByCategory(String category);
    List<Event> getEventsByLocation(String location);
    List<Event> getEventsByOrganizer(Long organizerId);
    List<Event> getEventsByTimeRange(LocalDateTime start, LocalDateTime end);
	List<Event> searchEvents(String category, String location, LocalDate date);
}

File: Frontend/src/app/services/event.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Event, EventSearchParams } from '../models/event.model';

@Injectable({ providedIn: 'root' })
export class EventService {

  private baseUrl = 'http://localhost:8081/events/events';

  constructor(private http: HttpClient) {}

  getAllEvents(): Observable<Event[]> {
    return this.http.get<Event[]>(this.baseUrl);
  }

  getEventById(id: number): Observable<Event> {
    return this.http.get<Event>(`${this.baseUrl}/${id}`);
  }

  createEvent(event: Partial<Event>): Observable<Event> {
    return this.http.post<Event>(this.baseUrl, event);
  }

  updateEvent(id: number, event: Partial<Event>): Observable<Event> {
    return this.http.put<Event>(`${this.baseUrl}/${id}`, event);
  }

  deleteEvent(id: number): Observable<string> {
    return this.http.delete(`${this.baseUrl}/${id}`, { responseType: 'text' });
  }

  searchEvents(params: EventSearchParams): Observable<Event[]> {
    let httpParams = new HttpParams();
    if (params.category) httpParams = httpParams.set('category', params.category);
    if (params.location) httpParams = httpParams.set('location', params.location);
    if (params.date)     httpParams = httpParams.set('date', params.date);
    return this.http.get<Event[]>(`${this.baseUrl}/search`, { params: httpParams });
  }

  getEventsByCategory(category: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.baseUrl}/category/${category}`);
  }

  getEventsByOrganizer(organizerId: number): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.baseUrl}/organizer/${organizerId}`);
  }

  getEventsByLocation(location: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.baseUrl}/location/${location}`);
  }

  getEventsByTimeRange(start: string, end: string): Observable<Event[]> {
    return this.http.get<Event[]>(`${this.baseUrl}/range`, {
      params: { start, end }
    });
  }

  triggerFeedback(eventId: number): Observable<string> {
    return this.http.post(`${this.baseUrl}/${eventId}/feedback`, {},
      { responseType: 'text' });
  }
}
